salsa-2022-tests/tests/lru.rs

```diff
@@ -103,3 +118,74 @@ fn lru_doesnt_break_volatile_queries() {
         }
     }
 }
+
+#[test]
+fn lru_can_be_changed_at_runtime() {
+    let mut db = Database::default();
+    assert_eq!(load_n_potatoes(), 0);
+
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 32);
+
+    get_hot_potato::set_lru_capacity(&db, 64);
+    assert_eq!(load_n_potatoes(), 32);
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 64);
+
+    // Special case: setting capacity to zero disables LRU
+    get_hot_potato::set_lru_capacity(&db, 0);
+    assert_eq!(load_n_potatoes(), 64);
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 192);
+
+    drop(db);
+    assert_eq!(load_n_potatoes(), 0);
+}
+
+#[test]
+fn lru_keeps_dependency_info() {
```

Comment:
I forgot to port this test in https://github.com/salsa-rs/salsa/pull/352 and ran into some issues with it.  

The old test used `synthetic_write`  to mock some input of durability has changed. we don't have this API right now
https://github.com/salsa-rs/salsa/blob/80d0d141945051ec86ecd979ac06cca1e364b27e/tests/lru.rs#L151  
https://github.com/salsa-rs/salsa/blob/80d0d141945051ec86ecd979ac06cca1e364b27e/src/runtime.rs#L100-L110

Reply:
Hmmm, I'm not sure why I didn't add this API, I think it's still on the runtime, giving warnings? I'd be ok to add it, though I'd keep it on the runtime.